hal/src/rtl872x/usbd_driver.cpp

```diff
@@ -263,8 +263,14 @@ int RtlUsbDriver::setupReply(SetupRequest* r, const uint8_t* data, size_t size)
             // some common cases of returning not a lot of data generated on stack
             memcpy(tempBuffer_, data, size);
             CHECK_RTL_USB_TO_SYSTEM(usbd_ep0_transmit(rtlDev_, tempBuffer_, size));
+#if MODULE_FUNCTION != MOD_FUNC_BOOTLOADER
+            HAL_Delay_Milliseconds(2);
```

Comment:
@YutingYou I think this just hides the problem and we can still run into it. Can we look into why the delay here is required? I would assume only two possible reasons:
1. Thread priority issues
2. Previous `usbd_ep0_transmit()` needs to complete which probably means we have to wait for `ep0DataInCb()` and queue things up

Reply:
@avtolstoy I tried several things before

1. ❌ improve the USB thread priority to the highest
2. ✅ HAL_Delay_Microseconds(2000)
3. ❌ Disable the scheduler
```
os_thread_scheduling(false nullptr);
HAL_Delay_Milliseconds(2);
os_thread_scheduling(true, nullptr);
```

So it might not be a multi-thread issue, I also confirmed `usbd_ep0_transmit ()` and `ep0DataInCb()` are running in the USB thread.

We're calling `usbd_ep0_transmit()` inside `ep0DataInCb()`, I'm not sure if this causes issues. In their CDC example, they don't event `use ep0_data_in` (we register this callback with `ep0DataInCb()`).
```
/* CDC ACM Class Driver */
usbd_class_driver_t usbd_cdc_driver = {
	.get_descriptor = cdc_acm_get_descriptor,
	.set_config = cdc_acm_set_config,
	.clear_config = cdc_acm_clear_config,
	.setup = cdc_acm_setup,
	.ep0_data_out = cdc_acm_handle_ep0_data_out,
	.ep_data_in = cdc_acm_handle_ep_data_in,
	.ep_data_out = cdc_acm_handle_ep_data_out,
};
```